### PR TITLE
Support for SHA1 and SHA384

### PR DESCRIPTION
**Pull request by joaoferreira-ctw**
Add support for SHA1 and SHA384
Add unit tests for new hashing algorithms and extend existing ones

### CODE REVIEW

File: src/mococrw/openssl_wrap.h [hunk 1]
```diff
@@ -556,7 +556,9 @@ void _X509_REQ_sign_ctx(X509_REQ* req, EVP_MD_CTX* ctx);
  * or hash computations.
  */
 enum class DigestTypes {
+    SHA1,
     SHA256, // @MARCUS: Shouldn't there be some value initialization with openssl consts here?
```
Comment (sz-ctw) [inline src/mococrw/openssl_wrap.h]: If remember correctly this @MARCUS comment is to be removed.

File: src/openssl_wrap.cpp [hunk 2]
```diff
@@ -418,10 +418,15 @@ void _X509_REQ_sign_ctx(X509_REQ* req, EVP_MD_CTX* ctx)
...
     switch (type) {
+        case DigestTypes::SHA1:
+            return lib::OpenSSLLib::SSL_EVP_sha1();
         case DigestTypes::SHA256:
             return lib::OpenSSLLib::SSL_EVP_sha256();
+        case DigestTypes::SHA384:
+            return lib::OpenSSLLib::SSL_EVP_sha384();
         case DigestTypes::SHA512:
             return lib::OpenSSLLib::SSL_EVP_sha512();
+
```
Comment (sz-ctw) [inline src/openssl_wrap.cpp]: Extra new line :yum:

File: tests/unit/openssl_lib_mock.cpp [hunk 3]
```diff
@@ -312,11 +312,21 @@ void OpenSSLLib::SSL_EVP_MD_CTX_destroy(EVP_MD_CTX* ptr) noexcept
...
+}
+
 const EVP_MD* OpenSSLLib::SSL_EVP_sha256() noexcept
 {
     return OpenSSLLibMockManager::getMockInterface().SSL_EVP_sha256();
 }
 
+const EVP_MD* OpenSSLLib::SSL_EVP_sha384() noexcept
+{
+    return OpenSSLLibMockManager::getMockInterface().SSL_EVP_sha512();
```
Comment (sz-ctw) [inline tests/unit/openssl_lib_mock.cpp]: Copy/Paste without replace :P Change 512 to 384 :P

File: tests/unit/test_hash.cpp [hunk 4]
```diff
@@ -33,38 +33,118 @@ class HashTest : public ::testing::Test
...
+{
+    EXPECT_THAT(utility::toHex(Hash::sha384().update("foo").digest()), Eq(sha384_foo));
+}
+
+TEST_F(HashTest, sha512SingleUpdateString)
+{
+    EXPECT_THAT(utility::toHex(Hash::sha512().update("foo").digest()), Eq(sha512_foo));
+}
+
+TEST_F(HashTest, sha1256MultipleUpdatesString)
```
Comment (sz-ctw) [inline tests/unit/test_hash.cpp]: Fix the name of the test maybe?

File: src/hash.cpp [hunk 5]
```diff
@@ -24,9 +24,23 @@ namespace mococrw
 {
 using namespace openssl;
 
+static const std::string SHA1 = "sha1";
```
Comment (sz-ctw) [inline src/hash.cpp]: Wasn't the story about sha384 only?

File: src/mococrw/openssl_wrap.h (hunk 1 above)
Comment (joaoferreira-ctw) [reply src/mococrw/openssl_wrap.h]: That comment was raised in another code review. Comment will be removed if requested in external review.

File: src/hash.cpp (hunk 5 above)
Comment (joaoferreira-ctw) [reply src/hash.cpp]: After external confirmation we ended up exposing SHA1 as well.

File: src/openssl_wrap.cpp (hunk 2 above)
Comment (joaoferreira-ctw) [reply src/openssl_wrap.cpp]: Removed.

File: tests/unit/openssl_lib_mock.cpp (hunk 3 above)
Comment (joaoferreira-ctw) [reply tests/unit/openssl_lib_mock.cpp]: Corrected.

File: tests/unit/test_hash.cpp (hunk 4 above)
Comment (joaoferreira-ctw) [reply tests/unit/test_hash.cpp]: Corrected.

**Comment by garw** [comment]
Manually squashed, rebased and merged

**Pull request closed without merging**